### ISSUE TITLE
[BUG] Runtime "Hang risk" error when launching 3DS via DropInComponent

### ISSUE THREAD

**Issue by ThomasCle** [Adyen/adyen-ios#1175]
**Describe the bug**
I get a ton of runtime errors regarding QoS thread issues when launching the 3D Secure 2 flow via the DropInComponent in SwiftUI:
```
This method should not be called on the main thread as it may lead to UI unresponsiveness.
Thread running at QOS_CLASS_USER_INTERACTIVE waiting on a lower QoS thread running at QOS_CLASS_DEFAULT. Investigate ways to avoid priority inversions
```

Xcode tracks some of them to this point of your code:
![Screenshot 2023-03-10 at 14 00 52](https://user-images.githubusercontent.com/1969572/224322584-e1118a95-2ec7-4842-91d1-a3516c10c695.png)

I am presenting the DropInComponent via your own SwiftUI ViewModifier for view controllers, which works fine in a lot of other cases. But when I use a 3DS test card it pops up with A LOT of run time errors. Everything works as expected, but Xcode is definitely not happy.

I have been pretty thorough using the `@MainActor` attribute when calling the SDK, where UI might be activated.

**To Reproduce**
1. Present the DropInComponent for Netherlands (NL) via SwiftUI.
2. Click on "Credit Card"
3. Type in "4917 6100 0000 0000" as test card.
4. See Xcode going crazy with run time errors!

**Expected behavior**
No runtime errors 🙈

**Screenshots**
-

**Smartphone (please complete the following information):**
 - Device: iPhone 14 Pro iOS Simulator
 - OS: iOS 16.2
 - SDK Version: 5.1.0

**Comment by descorp** [comment]
Hey @ThomasCle 

Thank you for reaching out!

It is indeed annoying popup 😅
In general, one can use 3DS2 safely and ignore this warning, it won't cause issues in production.

The cause has been fixed on [3DS lib](https://github.com/Adyen/adyen-3ds2-ios), but not yet released
We are planning to update v5.2.0 in Q2

**Comment by ThomasCle** [comment]
@descorp I appreciate your quick response and am grateful for the clarification of this issue in production. I'll look into the this again, once 5.2.0 is released. Thanks a lot!

**Comment by descorp** [comment]
@ThomasCle 

Released on [5.2.0](https://github.com/Adyen/adyen-ios/releases/tag/5.2.0)
Please confirm if this one doing the trick for you.

**Comment by descorp** [comment]
I am going to close this ticket for now.
Feel free to reopen or continue the discussion here.
